### ISSUE TITLE
Decode single image on Pytorch

### ISSUE THREAD

**Issue by liuwuliuyun** [NVIDIA/DALI#1508]
It seems that all DALI examples with Pytorch is used for model training. However, I'd like to use DALI when decode and resize a single test JPEG image and feed it to a Pytorch Network. Is there a way to not use Pipeline cuz it seems a bit of waste since I only decode and perfrom transform on a single image.

If Pipeline is necessary, what is the correct or recommended way to use it under this condition.

Thanks for any response.

**Comment by JanuszL** [comment]
Hi,
You can set the batch size to 1. DALI works best when you process multiple images at the time in batch as this fits best to the GPU.
The pipeline is the entity that hides the DALI processing graph under the hood so it is the way how DALI operates.
Can you tell us why you want to process only one image? In such a case maybe you don't need DALI?

**Comment by liuwuliuyun** [comment]
Thanks for your quick reply :)
I currently has a web service with pytorch model deployed. When user sends me an image, I need to decode this image and feed pytorch tensor to the model. But decoding image on CPU is slow. I first tried nvJPEG to accelarate the decoding process but had a very hard time converting decoded nvJPEG data structure in GPU memory to pytorch tensor. Then, I tired DALI. So in this case, should I use DALI?

**Comment by JanuszL** [comment]
Taking into account your goal DALI would be the easiest approach. Although it was designed and optimized with the training as the main use case, it is possible to use it in inference as well. Stil it maybe not as performant as some custom build solutions.
In your case, the easy of use would be the main advantage. What you should do is build a pipeline with only 1 thread and batch size of 1, with prefetch depth 1 - you want to process only one image at the time. Try to exploit the ExternalSource (or create some custom reader) so you can feed the data that user uploads.

**Comment by liuwuliuyun** [comment]
Thanks. I ll try
